webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/RuleApi.scala

```diff
@@ -944,7 +944,11 @@ class RuleApiService14 (
     def listCatAcc(categories: List[RuleCategory], acc: List[RuleCategoryId]) : List[RuleCategoryId] = {
       categories match {
         case Nil => acc
-        case c :: t => listCatAcc(t, c.id :: acc)
+        case c :: t =>
```

Comment:
you never explore the other subcategory if the head category is not empty

Reply:
should be something like this : 

val children =  listCatAcc(c.childs, c.id :: acc) 
listCatAcc(t, children)